index.js

```diff
@@ -79,6 +80,26 @@ class Schematic {
     const schem = sponge.write(this)
     return gzip(nbt.writeUncompressed(schem))
   }
+
+  /**
+   * similar to js map, loop over all schem blocks
+   * @param {(block: Block, pos: Vec3, index: number) => {}} callback
+   */
+  async map (callback) {
+    let counter = 0
+    const { startX, startY, startZ } = this.start()
+    const { endX, endY, endZ } = this.end()
+    for (let y = startY; y <= endY; y++) {
+      for (let z = startZ; z <= endZ; z++) {
+        for (let x = startX; x <= endX; x++) {
+          const pos = new Vec3(x, y, z)
+          const block = this.getBlock(new Vec3(x, y, z))
```

Comment:
Don't create the position twice, creating many objects can be costly.

For your information, when we loop over many blocks like this we usually try to reuse the same object for the position. Here it's best to create a new object because it's given to the callback and the user may store it, but otherwise it could be recycled.

Reply:
done